packages/cardano-services/src/Asset/CardanoTokenRegistry.ts

```diff
@@ -0,0 +1,56 @@
+import { Asset } from '@cardano-sdk/core';
+import { Logger } from 'ts-log';
+import { TokenMetadataService } from './types';
+import axios from 'axios';
+
+/**
+ * Dependencies that are need to create CardanoTokenRegistry
+ */
+export interface CardanoTokenRegistryDependencies {
+  logger: Logger;
+}
+
+/**
+ * TokenMetadataService implementation using Cardano Token Registry public API
+ */
+export class CardanoTokenRegistry implements TokenMetadataService {
+  #cache: Record<string, Asset.TokenMetadata> = {};
+  #logger: Logger;
+
+  constructor({ logger }: CardanoTokenRegistryDependencies) {
+    this.#logger = logger;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #checkNumberValue(data: any, attribute: string) {
+    if (attribute in data && typeof data[attribute].value === 'number') return data[attribute].value as number;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #checkStringValue(data: any, attribute: string) {
+    if (attribute in data && typeof data[attribute].value === 'string') return data[attribute].value as string;
+  }
+
+  async getTokenMetadata(subject: string): Promise<Asset.TokenMetadata | null> {
+    if (this.#cache[subject]) return this.#cache[subject];
+
+    this.#logger.debug(`Requesting for metatada for subject "${subject}"`);
+    const metadata: Asset.TokenMetadata = { version: '1.0' };
+
+    try {
+      const { data } = await axios.get(`https://tokens.cardano.org/metadata/${subject}`);
```

Comment:
1. The URL should also be passed in via the constructor.
2. Make a batch POST request like [here](https://github.com/input-output-hk/cardano-graphql/blob/ca5d556bcf0c2bd0b022ee3c293851b7ddd8f6f7/packages/api-cardano-db-hasura/src/MetadataClient.ts#L40-L50), also specifying the properties we know about.

https://input-output-hk.github.io/offchain-metadata-tools/api/latest/ -> `Batch Metadata Query`